aten/src/ATen/native/cuda/BinaryArithmeticKernel.cu

```diff
@@ -64,9 +64,38 @@ void mul_kernel_cuda(TensorIterator& iter) {
   }
 }
 
+void remainder_kernel_cuda(TensorIterator& iter) {
+  if (isIntegralType(iter.dtype(), /*includeBool*/ false)) {
+    AT_DISPATCH_INTEGRAL_TYPES(iter.dtype(), "remainder_cuda", [&]() {
+      using thrust_t = typename ztype_cuda<scalar_t>::thrust_t;
+      gpu_kernel_with_scalars(iter, []GPU_LAMBDA(thrust_t a, thrust_t b) -> thrust_t {
+        CUDA_KERNEL_ASSERT(b != 0);
```

Comment:
Hmmm, it seems that this device-side assert doesn't do what I was hoping. I wanted to create a catchable python exception, but that doesn't seem to happen:
```
>>> import torch
>>> a = torch.ones(1000, dtype=torch.int, device='cuda')
>>> b = torch.ones(1000, dtype=torch.int, device='cuda')
>>> b[500]=0
>>> try:
...     r = a.remainder(b)
... except RuntimeError:
...     print("error caught")
... 
>>> /home/kurtamohler/development/pytorch-remainder/aten/src/ATen/native/cuda/BinaryArithmeticKernel.cu:79: lambda [](int, int)->int::operator()(int, int)->int: block: [1,0,0], thread: [61,0,0] Assertion `b != 0` failed.
```
And interestingly, after running that, I can no longer do anything with cuda without quitting and rerunning python:
```
>>> torch.ones(1000, dtype=torch.int, device='cuda')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: CUDA error: device-side assert triggered
```
Notice that this now throws a python exception, but the exception is not thrown during the `remainder` call.
I'm not sure what the best solution would be.

Reply:
You can't do device side assert and get an assertion for this. You will have to give up on this until we land https://github.com/pytorch/pytorch/pull/26776/

I'd recommend just preserving the old behavior from CUDA for now

Reply:
Ah ok, sounds good to me.